beamer/agent.py

```diff
@@ -31,107 +37,165 @@ def _get_deployment_block(contract_info: dict[str, ContractInfo]) -> BlockNumber
 
-        source_request_manager = source_contracts["RequestManager"]
-        source_fill_manager = source_contracts["FillManager"]
+        if not source_chain.request_manager.functions.allowedLps(
+            self._config.account.address
+        ).call():
+            raise RuntimeError("Agent address is not whitelisted on RequestManager")
 
-        target_fill_manager = target_contracts["FillManager"]
-        target_request_manager = target_contracts["RequestManager"]
+    def _check_target_chain(self, target_chain: _Chain) -> None:
+        if not target_chain.fill_manager.functions.allowedLps(self._config.account.address).call():
+            raise RuntimeError("Agent address is not whitelisted on FillManager")
 
-        max_validity_period = source_request_manager.functions.MAX_VALIDITY_PERIOD().call()
+    def _setup_direction(
+        self, direction: tuple[ChainId, ChainId], chains: dict[ChainId, _Chain], l1: _BaseChain
+    ) -> None:
+        source_chain = chains[direction[0]]
+        target_chain = chains[direction[1]]
 
-        if self._config.unsafe_fill_time >= max_validity_period:
-            raise RuntimeError(f"Unsafe fill time must be less than {max_validity_period}")
-        if not target_fill_manager.functions.allowedLps(self._config.account.address).call():
-            raise RuntimeError("Agent address is not whitelisted on FillManager")
-        if not source_request_manager.functions.allowedLps(self._config.account.address).call():
-            raise RuntimeError("Agent address is not whitelisted on RequestManager")
+        self._check_source_chain(source_chain)
+        self._check_target_chain(target_chain)
 
-        claim_request_extension = source_request_manager.functions.claimRequestExtension().call()
+        claim_request_extension = (
+            source_chain.request_manager.functions.claimRequestExtension().call()
+        )
 
-        self.context = Context(
+        context = Context(
             requests=Tracker(),
             claims=Tracker(),
-            source_chain_id=source_chain_id,
-            target_chain_id=target_chain_id,
-            request_manager=source_request_manager,
-            fill_manager=target_fill_manager,
+            source_chain_id=source_chain.id,
+            target_chain_id=target_chain.id,
+            request_manager=source_chain.request_manager,
+            fill_manager=target_chain.fill_manager,
             match_checker=self._config.token_match_checker,
             address=self._config.account.address,
             latest_blocks={},
             config=self._config,
-            web3_l1=w3_l1,
+            web3_l1=l1.w3,
             task_pool=self._task_pool,
             claim_request_extension=claim_request_extension,
             l1_resolutions={},
             l1_invalidations={},
         )
+        direction_identifier = f"{direction[0]}->{direction[1]}"
```

Comment:
I think we should introduce more type-safety around the direction concept. For example, how about this:

```python

class TransferDirection(typing.NamedTuple):
     source: ChainID
     target: ChainID
```

If we had something like that in `beamer.typing`, it could be nicely used in functions currently taking plain `str` arguments. Additional benefit of having such a type is not having to use indices, but readable fields.

Note: we could also add a custom `__repr__` like

```python
def __repr__(self):
    return f"{self.source} -> {self.target}"
```
if we deem it necessary.